lib/acid/ffiutil.lua

```diff
@@ -0,0 +1,102 @@
+local ffi = require('ffi')
+local C = ffi.C
+local string_match = string.match
+local tableutil = require('acid.tableutil')
+
+local _M = { _VERSION = '1.0' }
+
+ffi.cdef [[
+long int strtol(const char *str, char **endptr, int base);
+unsigned long int strtoul(const char *nptr, char **endptr, int base);
+]]
+
+function _M.str_to_clong(str, mode)
+
+    if string_match(str, '^%-?%d+$') == nil then
+        return nil
```

Comment:
这里为啥要match一下先? 直接用`C.strtoul`转不行吗?

Reply:
非法数字转成nil啊

Reply:
C.strtoul 会把非法数字变成0

Reply:
恩...这里逻辑不太明显. 还有string match可能挺慢的.

然后我看man里写的不是0: `in the latter case, strtoul() returns ULONG_MAX and sets errno to ERANGE.  Precisely the same holds for strtoull() (with ULLONG_MAX instead of ULONG_MAX)`

Reply:
嗯，我再试试

Reply:
试了下：
strtoul 把'123abc'也认为是正确的输入，返回123；
'abc123'是错误输入，返回0，错误EINVAL；
'10000000000000000000000000000'会有ERANGE错误，返回ULONG_MAX。

http://www.cplusplus.com/reference/cstdlib/strtoul/ 这个标准里面没提'abc123'是否返回错误，只说了返回0.

str_to_clong 用的次数不是太多，慢点感觉还好。
我再添加下溢出也返回nil的情况。

Reply:
如果能拿到`errno`的话, 最好能像其他acid的lib一样, 第1个return value是结果, 第2, 3个是err code和err message